### DISCUSSION TITLE
Unable to detect schema changes using Databricks as DataSource

### DISCUSSION THREAD

**Discussion by craftanalytix** [TabularEditor/TabularEditor3#1347] [🙋♀️TE3 Q&A] [answered]
I have issue when trying to Update table schema of the table when my source is Databricks (Not Azure one)

<img width="498" alt="image" src="https://github.com/TabularEditor/TabularEditor3/assets/3468860/7c604a30-190a-4a8a-95d4-933c049c3ca2">

I can't figure out how to get this issue fixed.

**Comment by MariaJoseFF** [comment]
Hi @RoadDeal,

Thanks for reporting this. For us to reproduce this issue or help you further, we need further details:
- Where is your Databricks source hosted?
- Are you using an Implicit or a Structured data source? (see the [differences here](https://docs.tabulareditor.com/te3/tutorials/importing-tables.html#types-of-tom-data-sources))
- How do you import data from the table?  If you use a native query, there is a prerequisite: a default catalog must be specified when configuring the Databricks connection.

The Databricks connector is only available for Implicit data sources, which is why you’ll only see it on Power BI Datasets.
If you need to connect to Databricks in an Analysis Services model or using Provider or Structured data sources, you will need to use ODBC as usual.

**Reply by craftanalytix** [reply]
Hi @mjf-TabularEditor ,

- Databricks is hosted in AWS
- When using TE3 I am connecting using Implicit (this would work but the connection type when published is Azure Databricks)
- I did 3 scenarios to find out which one this warning occurs.

1. Connect/Import table using TE3 with Implicit and **it works just fine** but connection type in when creating a on-premise gateway will be Azure Databricks which not what I want as a connection type.
<img width="305" alt="image" src="https://github.com/user-attachments/assets/59886e22-7fea-4af5-86d0-fccadfc71427">

2. Doing it in Power BI Desktop and use Azure Databricks connector and published to workspace . After it published opening TE3 and connect to that deployed Model. This method **works fine** but still connection type is Azure Databricks and not what I want
<img width="339" alt="image" src="https://github.com/user-attachments/assets/db6fee99-3b66-482a-8772-c3f54b57d1fb">

3. Doing it again in Power BI Desktop and now using Databricks as connector and published into workspace. After it published, opening TE3 and connect to that model then perform "Update Table Schema". **This is where the error shows up and this is the connector we want to use.** 

<img width="317" alt="image" src="https://github.com/user-attachments/assets/83eddc39-2f60-40ed-a11e-53566918df6f">

<img width="312" alt="image" src="https://github.com/user-attachments/assets/0dd35215-01b3-47a6-9b81-58c87e9c4536">


Any guidance is greatly appreciated. Thank you!

**Comment by MariaJoseFF** [comment]
@RoadDeal, thanks for the details. I was able to reproduce the issue mentioned in point number 3.

When creating a Databricks connection in Power BI Desktop and publishing to the online workspace, Power BI uses another version of the Databricks connector that TE3 does not support. I will discuss these findings with the team and consider them in future releases.

If you need to update the schema of a table imported from a data source that is not supported by Tabular Editor 3, you can enable the Use Analysis Services for change detection option under Tools > Preferences > Schema Compare. You can read further details in our documentation: [Updating table schema through Analysis Services.](https://docs.tabulareditor.com/te3/tutorials/importing-tables.html#updating-table-schema-through-analysis-services).

I hope this helps.

**Comment by otykier** [comment, marked as answer]
Happy to let you know that [Tabular Editor v. 3.17.0](https://docs.tabulareditor.com/te3/other/release-notes/3_17_0.html?tabs=downloads) now has support for Databricks on non-Azure hosts, i.e. the `DatabricksMultiCloud.Catalogs` M-function.

**Reply by craftanalytix** [reply]
This is amazing news. Thank you @otykier and TE team
